test/unit_test/runtime/magnitude_test.cpp

```diff
@@ -218,8 +287,50 @@ TEST_CASE("Can raise Magnitudes to rational powers")
   }
 }
 
+TEST_CASE("can distinguish integral, rational, and irrational magnitudes")
+{
+  SECTION("Integer magnitudes are integral and rational")
+  {
+    auto check_rational_and_integral = [](Magnitude auto m) {
+      CHECK(is_integral(m));
+      CHECK(is_rational(m));
+    };
+    check_rational_and_integral(as_magnitude<1>());
```

Comment:
I recall from the first PR that `magnitude<>` is valid. Why not test that, too? It seems to be integral and rational due to the use of fold expressions in the implementation.

Reply:
Done.